src/controllers/dlgprefcontroller.cpp

```diff
@@ -557,9 +558,22 @@ void DlgPrefController::slotApply() {
         return;
     }
 
-    QString mappingPath = mappingPathFromIndex(m_ui.comboBoxMapping->currentIndex());
+    QString mappingFilePath = mappingFilePathFromIndex(m_ui.comboBoxMapping->currentIndex());
+
+    auto mappingFileInfo = QFileInfo(mappingFilePath);
+
+    if (m_pMapping) {
+        m_pMapping->saveSettings(mappingFileInfo, m_pConfig, m_pController->getName());
```

Comment:
I do not understand why we save here a setting. For me it looks like we save the new data to the old mapping, which would be a bug. But maybe only an explaining comment is missing.

Reply:
Let me know if that comment make sense, happy to reformulate otherwise! The fact that the state management occurs across those 3 slots is certainly puzzling, but when I started working on this feature, words were that preferences would soon be tear down and rebuild using QML so I didn't want to spend to much effort on cleaning this up

Reply:
I have refactored the flow so `apply` slot only `save` and `show` slot `load`.